### ISSUE TITLE
Validator can fail to parse valid NIfTI file when header is not contained in first 500 bytes of compressed file

### ISSUE THREAD

**Issue by mrserody** [bids-standard/legacy-validator#1135]
I am trying to upload a dataset onto OpenNeuro; however, I keep on receiving a validation error that OpenNeuro is unable to parse the header data for one subject's T2w file. I can view the file just fine on FSL and on OpenNeuro, so I am pretty sure this is a bug.

**Comment by effigies** [comment]
tl;dr. I reproduced the issue, and found that the root cause is that the first 500 bytes of this compressed file do not yield 348 uncompressed bytes. I have two recommendations:

1) @mrserody In the short term, you can use a higher compression ratio and ensure the original filename is not stored, to maximize the amount of uncompressed data that can be recovered from the start of the file. You can use the following commands:

```
gunzip sub-C43/anat/sub-C43_rec-1_T2w.nii.gz
gzip -9 -n sub-C43/anat/sub-C43_rec-1_T2w.nii
```

2) We should increase the number of uncompressed bytes read to ensure that the full header can be read. Or we should find a way to read uncompressed bytes into the buffer. 1KiB seems like a safely large buffer.

---

I can reproduce the issue on the [web validator](https://bids-standard.github.io/bids-validator/) using https://openneuro.org/datasets/ds003404/file-display/sub-C43:anat:sub-C43_rec-2_T1w.nii.gz and patching it into a fake dataset.

If I `gunzip` the file, validation passes, and the file is still found. If I re-`gzip` the file (no options), the error returns.

It looks like 500 bytes are read from the compressed or uncompressed file:

https://github.com/bids-standard/bids-validator/blob/b8e11de16738b66d67c316e4d9040ed1c1c1dbcf/bids-validator/utils/files/readNiftiHeader.js#L44-L72

If <348 uncompressed bytes are extractable from those 500 bytes, then the header will be incomplete, which could produce the issue.

To verify, here's some Python:

```Python
import gzip, io
with open("sub-C43_rec-1_T2w.nii.gz", "rb") as fobj:
    bstr = fobj.read(2048)

gz = gzip.GzipFile(fileobj=io.BytesIO(bstr[:500]), mode='rb')
for i in range(348):
    try:
        gz.read(1)
    except EOFError:
        break
print(f"Read {i} bytes")
```

Yields: `Read 345 bytes`

I tried re-gzipping with `-n` (do not store original filename) and `-9` (highest compression):

```
gunzip sub-C43/anat/sub-C43_rec-1_T2w.nii.gz
gzip -9 -n sub-C43/anat/sub-C43_rec-1_T2w.nii
```

This resulted in a successful validation.

**Comment by effigies** [comment]
This will be fixed in the next release. @rwblair Let's make sure to get a patch release that can be included in the next OpenNeuro deployment.

**Comment by effigies** [comment]
This has been resolved in the latest release. Thank you very much for reporting!
